src/validators.ts

```diff
@@ -96,11 +99,171 @@ export function length(
 }
 
 // regex
-// minNum
-// maxNum
-// dateFormat
-// minDate
-// maxDate
+export function pattern(
+  option: RegExp,
+  errorMsg?: CustomErrMsg<string, RegExp>
+){
+  return(val: string, context?: Context): string | undefined => {
+    const defaultErrorMsg = `The value ${val} doen't match the pattern ${option}`;
+    return option.test(val)
+      ?undefined
+      :getError({
+        errorMsg,
+        defaultErrorMsg,
+        val,
+        context,
+        option,
+      });
+  }
+}
+
+ // validator for minimum no of digits in a number
+export function min(
+  option: number,
+  errorMsg?: CustomErrMsg<number, number>,
+) {
+  return (val: number, context?: Context): string | undefined => {
+    const defaultErrorMsg = `The entered value: ${val} should be at greater than ${option}`;
+    return val < option
+      ? getError({ errorMsg, defaultErrorMsg, val, context, option })
+      : undefined 
+  }
+}
+
+ // validator for maximum no of digits in a number
+export function max(
+  option: number,
+  errorMsg?: CustomErrMsg<number, number>,
+) {
+  return (val: number, context?: Context): string | undefined => {
+    const defaultErrorMsg = `The entered value: ${val} should be smaller than ${option}`;
+    return val > option
+      ? getError({ errorMsg, defaultErrorMsg, val, context, option })
+      : undefined 
+  }
+}
+
+  // parses day, month and year from the given date
+function parseDate(date: Date): number[] {
+  const months = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"];
+  
+  const parts = date.toString().split(" ");
+  const day = parseInt(parts[2], 10);
+  const month = months.indexOf(parts[1]) + 1;
+  const year = parseInt(parts[3], 10);
+  
+  const dateValue:number[] = [day, month, year];
+  return dateValue;
+}
+
+ // dateFormat
+export function validateDate(
```

Comment:
We need some love to the date decoder. Its located here https://github.com/ludbek/validatex/blob/master/src/decoders.ts#L170

Its implementation is wrong at the moment. It should take an unknown value and convert it to a date. The expected unknown values will be either a string or a date itself. Try moving these logics to there.

Also we might not need to parse the date string manually. Javascript is already capable of parsing an ISO format dates. Checkout https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse